app/controllers/chargeback_assignment_controller.rb

```diff
@@ -4,110 +4,83 @@ class ChargebackAssignmentController < ApplicationController
-    replace_right_cell
+  def index
+    @title = _("Chargeback Assignments")
+    @tabform = "Compute"
+    session[:changed] = @changed = false
+    build_tabs
+    set_form_vars
+    render :action => "show"
   end
 
-  def explorer
-    @breadcrumbs = []
-    @explorer    = true
-    build_accordions_and_trees
-
-    @right_cell_text = _("All Assignments")
-    set_form_locals if @in_a_form
-    session[:changed] = false
-
-    render :layout => "application" unless request.xml_http_request?
-  end
-
-  def set_form_locals
-    @x_edit_buttons_locals = {
-      :action_url   => 'cb_assign_update',
-      :no_cancel    => true,
-      :multi_record => true
-    }
+  def change_tab
+    clear_flash_msg
+    @tabform = params['uib-tab']
+    build_tabs
+    set_form_vars
+    render :action => "show"
   end
 
   # AJAX driven routine to check for changes in ANY field on the form
-  def cb_assign_field_changed
-    return unless load_edit("cbassign_edit__#{x_node}", "replace_cell__chargeback")
-
-    cb_assign_get_form_vars
+  def form_field_changed
+    return unless load_edit("cbassign_edit__#{params[:id]}", "index")
+    get_form_vars
     render :update do |page|
       page << javascript_prologue
       except = %i[cbshow_typ cbtag_cat cblabel_key]
       changed = (@edit[:new].except(*except) != @edit[:current].except(*except))
-      page.replace("cb_assignment_div", :partial => "cb_assignments") if params[:cbshow_typ] || params[:cbtag_cat] || params[:cblabel_key]
+      prefix = @edit[:new][:type].downcase
+      page.replace("#{prefix}_assignments_div", :partial => "#{prefix}_assignments") if params[:cbshow_typ] || params[:cbtag_cat] || params[:cblabel_key]
       page << javascript_for_miq_button_visibility(changed)
     end
   end
 
-  def cb_assign_update
+  def update
+    clear_flash_msg
+    return unless load_edit("cbassign_edit__#{params[:id]}", "index")
     if params[:button] == "reset"
-      get_node_info(x_node)
-      add_flash(_("All changes have been reset"), :warning)
-      replace_right_cell
-    else
-      return unless load_edit("cbassign_edit__#{x_node}", "replace_cell__chargeback")
-
-      cb_assign_set_record_vars
-      rate_type = x_node.split('-').last
+      set_form_vars
+      flash_to_session(_("All changes have been reset"), :warning)
+    elsif params[:button] == "save"
+      set_record_vars
+      rate_type = params[:id]
       begin
         ChargebackRate.set_assignments(rate_type, @edit[:set_assignments])
       rescue StandardError => bang
         render_flash(_("Error during 'Rate assignments': %{error_message}") % {:error_message => bang.message}, :error)
       else
-        add_flash(_("Rate Assignments saved"))
-        get_node_info(x_node)
-        replace_right_cell
+        flash_to_session(_("Rate Assignments saved"))
       end
+    else
+      flash_to_session("Rate Assignment has been cancelled")
+    end
+    set_form_vars unless params[:button] == 'reset'
```

Comment:
Can we somehow move this up to if-else, so there's no need for additional conditions?